src/main/scala/tilelink/RegisterRouter.scala

```diff
@@ -173,7 +181,7 @@ class TLRegisterRouter[B <: TLRegBundleBase, M <: LazyModuleImp](
 
 /** Mix this trait into a RegisterRouter to be able to attach its register map to a TL bus */
 trait HasTLControlRegMap { this: RegisterRouter[_] =>
-  protected val controlNode = TLRegisterNode(
+  val controlNode = TLRegisterNode(
```

Comment:
I don't like that you had to remove this protection! That means you are making the API leakier. Why isn't `getMemory` a function here that accesses the node's device?